nucypher/device/trezor.py

```diff
@@ -0,0 +1,178 @@
+
+        sig = trezor_eth.sign_message(self.client, bip44_path, message)
+        return self.Signature(sig.signature, sig.address)
+
+    @_handle_device_call
+    @validate_checksum_address
+    def verify_message(self, signature: bytes, message: bytes, checksum_address: str):
+        """
+        Verifies that a signature and message pair are from a specified
+        address via the TREZOR ethereum verify_message API. This method
+        requires interaction between the TREZOR and the user.
+
+        If the signature or message is not valid, it will raise a
+        nucypher.crypto.signing.InvalidSignature exception. Otherwise, it
+        will return True.
+        """
+        is_valid = trezor_eth.verify_message(self.client, checksum_address,
+                                             signature, message)
+        if not is_valid:
+            raise InvalidSignature("Signature verification failed.")
+        return True
+
+    @_handle_device_call
+    def get_address(self, address_index: int = None, hd_path: str = None):
+        """
+        Derives an address available on the Trezor via the ethereum
+        get_address API and returns it.
+        """
+        if address_index is not None and hd_path is None:
+            hd_path = self.ETH_BIP44_PATH.format(address_index=address_index)
+            hd_path = trezor_tools.parse_path(hd_path)
+        elif address_index is None and hd_path is not None:
+            hd_path = trezor_tools.parse_path(hd_path)
+        else:
+            raise ValueError("You must provider either an address_index or an hd_path.")
+
+        address = trezor_eth.get_address(self.client, hd_path)
+        return address
+
+    @_handle_device_call
+    def sign_eth_transaction(self, checksum_address: str, transaction: dict) -> bytes:
+        """
+        Signs an Ethereum transaction via the Trezor ethereum sign_tx API
+        and returns the signed, RLP encoded transaction.
+
+        TODO: Is there any input validation required for the transaction
+              data that is passed in?
+        """
+        try:
+            bip44_path = self.__addresses[checksum_address]
+        except KeyError:
+            raise self.DeviceError(f'{checksum_address} is not available as a cached address on this device.')
+
+        # Handle Web3.py -> trezor API formatting
+        assert_valid_fields(transaction)
+        transaction['to'] = to_canonical_address(transaction['to'])
+        trezor_tx_keys = {'gas': 'gas_limit',
+                          'gasPrice': 'gas_price',
+                          'chainId': 'chain_id'}
+        trezor_format_tx = dict(apply_key_map(trezor_tx_keys, transaction))
+
+        tx_v, tx_r, tx_s = trezor_eth.sign_tx(client=self.client, n=bip44_path,
+                                              **trezor_format_tx)
+
+        # The chain_id is not necessary to include, the Trezor API creates an
+        # EIP-155 signature if needed.
+        del transaction['chainId']
```

Comment:
Additionally, we can take a non-mutative approach to manipulating transaction `dict` values:

```
from cytoolz.dicttoolz import dissoc

unsigned_transaction = dissoc(unsigned_transaction, 'chainId')
```